src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/loadbalancer/ApacheEc2LoadBalancerFormula.kt

```diff
@@ -82,25 +86,48 @@ internal class ApacheEc2LoadBalancerFormula : LoadBalancerFormula {
     }
 }
 
-private class ApacheLoadBalancer(
+private class HaProxyLoadBalancer(
     private val nodes: List<URI>,
     private val ssh: Ssh,
     private val httpPort: Int
 ) : LoadBalancer {
 
-    init {
-        throw Exception("Wait until the `infrastructure` module ships `ApacheLoadBalancer`")
-    }
-
     override val uri: URI = URI("http://${ssh.host.ipAddress}:$httpPort/")
 
     override fun waitUntilHealthy(
         timeout: Duration
     ) {
-        throw Exception(
-            "TODO: Install Apache load balancer via $ssh," +
-                " configure it to balance between $nodes" +
-                " and listen on $httpPort"
-        )
+
     }
+
+    private val APACHE_CONFIG_PATH = "/etc/apache2/sites-enabled/000-default.conf"
+
+    fun provision() {
+        ssh.newConnection().use { connection ->
+            connection.safeExecute("sudo apt-get update", Duration.ofMinutes(3))
+            connection.safeExecute("sudo apt-get --assume-yes install apache2", Duration.ofMinutes(1))
```

Comment:
We can reuse `Ubuntu().install` for a few improvements, e.g. error checking, retries, race-condition control.

Reply:
Done.